javascript/packages/fury/lib/gen/router.ts

```diff
@@ -17,28 +17,22 @@
  * under the License.
  */
 
-import { Fury } from "../type";
-import { InternalSerializerType, RefFlags } from "../type";
+import { TypeDescription } from "../description";
+import { SerializerGenerator } from "./serializer";
+import { InternalSerializerType } from "../type";
+import { Builder } from "./builder";
+import { Scope } from "./scope";
 
-export default (fury: Fury) => {
-  const { binaryReader, binaryWriter, referenceResolver } = fury;
-  const { uint8: readUInt8 } = binaryReader;
-  const { int8: writeInt8, uint8: writeUInt8 } = binaryWriter;
-  return {
-    ...referenceResolver.deref(() => {
-      return readUInt8() === 0 ? false : true;
-    }),
-    write: referenceResolver.withNotNullableWriter(InternalSerializerType.BOOL, false, (v: boolean) => {
-      writeUInt8(v ? 1 : 0);
-    }),
-    writeWithoutType: (v: boolean) => {
-      writeInt8(RefFlags.NotNullValueFlag);
-      writeUInt8(v ? 1 : 0);
-    },
-    config: () => {
-      return {
-        reserve: 4,
-      };
-    },
-  };
-};
+type SerializerGeneratorConstructor = new (description: TypeDescription, builder: Builder, scope: Scope) => SerializerGenerator;
+
+export class Register {
```

Comment:
This can be named as `Registry`, so `reg` can be renamed to `register`

Reply:
Seems it's codegen related, maybe we can name it as `CodegenRegistry` or something like

Reply:
In java we have a class named as `CodeGenerator`, and has a class named `CodecBuilder`. Hope it can give some inspiration